### PR TITLE
[Meru][Easier Onboarding] Invitations API handles existing members/invitations

### PR DESCRIPTION
**Pull request by philipperolet**
Description
---
Context: this is part of task #4419

Following discussions with ed, the invitations API is changed as follows:
- if an invitation is sent for an existing member (revoked or not) => return error (cannot send invitation to existing)
- if there is an existing pending invitation, mark it as revoked & create a new one, so that we re-send an email

Deploy
---
Deploy front

### CODE REVIEW

**Comment by lasryaric** [comment]
> if an invitation is sent for an existing member (revoked or not) => return error (cannot send invitation to existing)

So you can never invite someone who has already been invited on the workspace? Why is that?

**Comment by philipperolet** [comment]
> > if an invitation is sent for an existing member (revoked or not) => return error (cannot send invitation to existing)
> 
> So you can never invite someone who has already been invited on the workspace? Why is that?

Inviting someone who's already invited is possible, and is done (see line below)

If you meant "you can never invite someone who is an existing member" => indeed, you can't with this endpoint since they're already there so the invitation flow creating a user, a membership, etc. does not make sense. 

The invariant is that the endpoint is to send invites to people who've never been to the workspace and don't have an existing membership
 
However, you're right that we want to be able to re-add members that were revoked. This is handled via the `members` endpoint, we set back their role from 'revoked' to 'member' or other. (and this is handle by the UI)

Note: this PR does not change behaviour in this respect. Previous code prevented api calls with existing members at UX sites

File: front/lib/api/invitation.ts [hunk 1]
```diff
@@ -25,29 +25,75 @@ function typeFromModel(
   };
 }
 
-export async function createInvitation(
+export async function createOrRecreateInvitation(
```
Comment (lasryaric) [inline front/lib/api/invitation.ts:28]: per IRL conversation, call this function `getOrCreaateInvitation()` and do not revoke pending invitations when one already exists.

**Review by lasryaric — CHANGES_REQUESTED** [review]

File: front/lib/api/invitation.ts (hunk 1 above)
Comment (philipperolet) [reply front/lib/api/invitation.ts]: AS per our re-discussion, ok :)

**Comment by philipperolet** [comment]
@lasryaric applied your com, PMRW

**Pull request merged**